hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/io/asyncfs/FanOutOneBlockAsyncDFSOutput.java

```diff
@@ -354,10 +364,24 @@ private void setupReceiver(int timeoutMs) {
     }
   }
 
+  private static final String DUMMY_DFS_OUTPUT_STREAM_CLASS =
+    "org.apache.hadoop.hdfs.DummyDFSOutputStream";
+
+  private DFSOutputStream tryCreateDummyDFSOutputStream(EnumSet<CreateFlag> createFlags) {
+    try {
+      return (DFSOutputStream) Class.forName(DUMMY_DFS_OUTPUT_STREAM_CLASS).getConstructors()[0]
+        .newInstance(this, client, src, stat, createFlags, summer);
+    } catch (Exception e) {
+      LOG.debug("can not instantiate DummyDFSOutputStream, should be hadoop 2.x", e);
```

Comment:
This would log on every file open, unlike how we handle the Hadoop-version depndent logging in other cases.

Reply:
We could have a try to safe the constructor so we do not need to do this everytime.